### PR TITLE
Update modules required for rnaseq pipeline

### PR DESCRIPTION
**Pull request by drpatelh**
Udpdating modules required for the nf-core/rnaseq pipeline

* Fix memory options in hisat2/build module
* Bump Preseq version and added `meta.yml` to get lint tests passing
* Add ability to auto-detect `--genomeSAindexNbases` for smaller genomes in star/genomegenerate module

### CODE REVIEW

**Comment by drpatelh** [comment]
`hisat2/build` tests are expected to fail here because we have had to rename the test directory to [`hisat2/build_test`](https://github.com/nf-core/modules/tree/master/tests/software/hisat2/build_test) to get pytest to work.

**Comment by drpatelh** [comment]
Ready for review @JoseEspinosa @KevinMenden

**Review by KevinMenden — APPROVED** [review]
LGTM! :+1: 
Bit annoying that we need three different tools now in the STAR module but what can you do :shrug:

File: software/hisat2/build/main.nf [hunk 1]
```diff
@@ -42,7 +42,7 @@ process HISAT2_BUILD {
     def ss = ''
     def exon = ''
     def extract_exons = ''
-    def hisat2_build_memory = (params.hisat2_build_memory as nextflow.util.MemoryUnit).toGiga()
+    def hisat2_build_memory = params.hisat2_build_memory ? (params.hisat2_build_memory as nextflow.util.MemoryUnit).toGiga() : 0
```
Comment (JoseEspinosa) [inline software/hisat2/build/main.nf:45]: 🤟

**Review by JoseEspinosa — APPROVED** [review]
Nice!

**Comment by drpatelh** [comment]
> Bit annoying that we need three different tools now in the STAR module but what can you do

Yup, but this will save alot of failures being reported for the rnaseq pipeline and others. Those additional tools should have quite a small storage footprint too. I will compare the size of the containers and post here

**Comment by drpatelh** [comment]
Yup, almost 4 times the size 😏 But still much smaller than some of those Java beasts!

```console
8.9M	depot.galaxyproject.org-singularity-star-2.6.1d--0.img
37M	depot.galaxyproject.org-singularity-mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2-59cdd445419f14abac76b31dd0d71217994cbcc9-0.img
```

**Comment by KevinMenden** [comment]
Yeah that's still really small! 👍

**Pull request merged**